orders/views.py

```diff
@@ -86,10 +99,99 @@ def post(self, request):
                 item                            = Item.objects.get(id=order_item.item_id)
                 item.order_quantity             += order_item.quantity
                 item.stock                      -= order_item.quantity               
+                
+                if Cart.objects.filter(Q(member_id=request.user.id) & Q(item_id = order_item.item_id)).exists():
+                    Cart.objects.filter(Q(member_id=request.user.id) & Q(item_id = order_item.item_id)).delete()
+
                 order_item.save()
                 item.save()
 
             return JsonResponse({'MESSAGE': "SUCCESS"}, status=201)
         
         except KeyError:
-            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
\ No newline at end of file
+            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
+
+
+
+
+class CartView(View):
+    @login
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if Cart.objects.filter(Q(item_id = data['item_id']) & Q(member_id=request.user.id)).exists():
+                return JsonResponse({"MESSAGE":"CART_EXIST"}, status=400)
+            
+            if data['quantity'] > Item.objects.get(id = data['item_id']).stock:
+                return JsonResponse({"MESSAGE":"NO_STOCK"}, status=400)
+                
+            Cart.objects.create(
+                item_id=data['item_id'], 
+                member_id = request.user.id, 
+                quantity = data['quantity']
+            )
+            return JsonResponse({"MESSAGE" : "SUCCESS"}, status=201)
+
+        except KeyError:
+            return JsonResponse({"MESSAGE": "KEY_ERROR"}, status=400)
+
+    @login
+    def get(self, request):
+
+        if not Cart.objects.filter(member_id=request.user.id).exists():
+                return JsonResponse({"MESSAGE":"EMPTY_CART"}, status=400)
+
+        results = [
+            {
+            "cart_id"  : cart.id,
+            "item_id"  : cart.item.id,
+            "quantity" : cart.quantity,
+            "name"     : cart.item.name,
+            "image"    : cart.item.image_set.get(main=1).image_url,
+            "price"    : cart.item.price,
+            "discount" : cart.item.discount
+            } 
+        for cart in Cart.objects.filter(member_id=request.user.id)]
+
+        return JsonResponse({"RESULT" : results}, status=201)
```

Comment:
조회, 업데이트에 사용되는 status code는 200입니다.

```suggestion
    @login
    def get(self, request):

        if not Cart.objects.filter(member_id=request.user.id).exists():
                return JsonResponse({"MESSAGE":"EMPTY_CART"}, status=400)

        results = [
            {
            "cart_id"  : cart.id,
            "item_id"  : cart.item.id,
            "quantity" : cart.quantity,
            "name"     : cart.item.name,
            "image"    : cart.item.image_set.get(main=1).image_url,
            "price"    : cart.item.price,
            "discount" : cart.item.discount
            } 
        for cart in Cart.objects.filter(member_id=request.user.id)]

        return JsonResponse({"RESULT" : results}, status=200)
```

Reply:
나머지는 👍